### DISCUSSION TITLE
Constant macroparticle weight injection at a point

### DISCUSSION THREAD

**Discussion by binahn91** [BLAST-WarpX/warpx#5495] [Q&A]
Dear developers,

I am having trouble with **constant MW(Macroparticle Weight) continuous particle injection at a single point(or node)**.

I would like to perform both ES and EM PIC for a RZ system,
with **initially empty space**,
**a point source(invisible cathode)** at the center,
which is **emitting energetic electrons**,
**isotropically** at a **constant rate**.

I tried `NFluxPerCell`,
but it is hard to figure out
how to adjust `<species_name>.flux` for **constant rate emission**, **isotropy** and **constant MW**.
(No varying MW)

My goal is to investigate the confinement of single charged species (non-neutral plasma) in various E B field, in a lab-scale chamber.

I am trying things by referring to examples such as [inputs_test_rz_flux_injection](https://github.com/ECP-WarpX/WarpX/blob/development/Examples/Tests/flux_injection/inputs_test_rz_flux_injection), but to no avail.

Maybe, should I use the '`<species_name>.do_continuous_injection` with `warpx.do_moving_window=1`' approach
like [issues5131](https://github.com/ECP-WarpX/WarpX/issues/5131) ?

Thank you for your help!
Sincerely,
Bin Ahn

**Comment by roelof-groenewald** [comment]
Hi @binahn91.  

I think you should be able to do what you are interested in with the `NFluxPerCell` injector. The one issue is that you won't be able to inject from a single point, but will need to inject over a small area - this can be controlled with the `flux_profile` input. For example, setting:
```
<species_name>.flux_profile = "(x*x + y*y < 0.1)
<species_name>.surface_flux_pos = 0
<species_name>.momentum_distribution_type = 'gaussian'
```
should continuously inject particles over the disc r < 0.1 with a Gaussian velocity distribution.
Will you give that a try and let us know if you encounter any further issues?

**Comment by binahn91** [comment]
Thank you @roelof-groenewald 
I actually partially solved this issue earlier, like the following, similar to your suggestion.

```
############################################################
################################## [Particle initialization]

particles.species_names = elec
elec.species_type = electron

elec.xmin =  0    
elec.xmax =  0.001  
elec.zmin =  0.049
elec.zmax =  0.051

elec.injection_sources = inject0 inject1

elec.inject0.injection_style = NFluxPerCell
elec.inject0.flux_normal_axis = r
elec.inject0.surface_flux_pos = 0.0001
elec.inject0.flux_direction = +1
elec.inject0.num_particles_per_cell = 1.e1
elec.inject0.flux_profile = constant
elec.inject0.flux = 1e20
elec.inject0.momentum_distribution_type = constant
elec.inject0.ux = 0.186
elec.inject0.uz = 0.186
```
This makes the emission **constant in time** and **monoenergetic**, as I wanted.
However, making it **isotropic** was not very simple.
For now, I just assigned many `injection_sources` and gave different directions for pseudo-isotropy.

I will try @roelof-groenewald 's suggestion.

I am also hoping to easily adjust constant particle weight,
like the `MultipleParticles` method, but with certain rate of injection per time.
(not determined by `num_particles_per_cell`, `flux`, etc.)

So, I may try to implement this myself.

I will share the result here in the future.
